packages/passport/sdk/src/zkEvm/zkEvmProvider.ts

```diff
@@ -219,11 +220,29 @@ export class ZkEvmProvider implements Provider {
           method: request.method,
           params: request.params || [],
           ethSigner,
-          jsonRpcProvider: this.#jsonRpcProvider,
+          staticJsonRpcProvider: this.#staticJsonRpcProvider,
           relayerClient: this.#relayerClient,
           guardianClient: this.#guardianClient,
         });
       }
+      case 'eth_chainId': {
+        // Call detect network to fetch the chainId so to take advantage of
+        // the caching layer provided by StaticJsonRpcProvider.
+        // In case Passport is changed from StaticJsonRpcProvider to a
+        // JsonRpcProvider, this function will still work as expected given
+        // that detectNetwork call _uncachedDetectNetwork which will force
+        // the provider to re-fetch the chainId from remote.
+        return new Promise<string>((resolve, reject) => {
+          (async () => {
+            try {
+              const { chainId } = await this.#staticJsonRpcProvider.detectNetwork();
+              resolve(utils.hexlify(chainId));
+            } catch (err) {
+              reject(err);
+            }
+          })();
```

Comment:
What's the reason for wrapping this in a Promise & a try/catch? This can be simplified to the following without any functional difference, can't it?

```suggestion
        const { chainId } = await this.#staticJsonRpcProvider.detectNetwork();
        return utils.hexlify(chainId);
```